src/KahanSummation.jl

```diff
@@ -29,6 +29,59 @@ else
     promote_sys_size_add(x::T) where {T} = Base.r_promote(+, zero(T)::T)
 end
 
+
+import Base.TwicePrecision
+
+
+function plus_kbn(x::T, y::T) where {T}
+    hi = x + y
+    lo = abs(x) > abs(y) ? (x - hi) + y : (y - hi) + x
+    TwicePrecision(hi, lo)
+end
+function plus_kbn(x::T, y::TwicePrecision{T}) where {T}
+    hi = x + y.hi
+    if abs(x) > abs(y.hi)
+        lo = ((x - hi) + y.hi) + y.lo
+    else
+        lo = ((y.hi - hi) + x) + y.lo
+    end
+    TwicePrecision(hi, lo)
+end
+plus_kbn(x::TwicePrecision{T}, y::T) where {T} = plus_kbn(y, x)
+
+function plus_kbn(x::TwicePrecision{T}, y::TwicePrecision{T}) where {T}
+    hi = x.hi + y.hi
+    if abs(x.hi) > abs(y.hi)
+        lo = (((x.hi - hi) + y.hi) + y.lo) + x.lo
+    else
+        lo = (((y.hi - hi) + x.hi) + x.lo) + y.lo
+    end
+    TwicePrecision(hi, lo)
+end
+
+Base.r_promote_type(::typeof(plus_kbn), ::Type{T}) where {T<:AbstractFloat} =
+    TwicePrecision{T}
+
+Base.mr_empty(f, ::typeof(plus_kbn), T) = TwicePrecision{T}
+
+singleprec(x::TwicePrecision{T}) where {T} = convert(T, x)
+
+
+"""
+    sum_kbn([f,] A)
+
+Return the sum of all elements of `A`, using the Kahan-Babuska-Neumaier compensated
+summation algorithm for additional accuracy.
+"""
+sum_kbn(f, X) = singleprec(mapreduce(f, plus_kbn, X))
+sum_kbn(X) = sum_kbn(identity, X)
+
+
+
+
+
+
+
```

Comment:
This is a lot of blank lines. Could you reduce it to just one, maybe two?

Reply:
No. I need those for caching :trollface: